sdk/js-query/src/query/ethCall.ts

```diff
@@ -0,0 +1,48 @@
+import { BinaryWriter } from "./BinaryWriter";
+import { HexString } from "./consts";
+import { ChainQueryType, ChainSpecificQuery } from "./request";
+import { hexToUint8Array, isValidHexString } from "./utils";
+
+export interface EthCallData {
+  to: string;
+  data: string;
+}
+
+// Can be a block number or a block hash
+export type BlockTag = number | HexString;
+
+export class EthCallQueryRequest implements ChainSpecificQuery {
+  blockTag: string;
+
+  constructor(blockTag: BlockTag, public callData: EthCallData[]) {
+    if (typeof blockTag === "number") {
+      this.blockTag = `0x${blockTag.toString(16)}`;
+    } else if (isValidHexString(blockTag)) {
+      if (!blockTag.startsWith("0x")) {
+        blockTag = `0x${blockTag}`;
+      }
+      this.blockTag = blockTag;
+    } else {
+      throw new Error(`Invalid block tag: ${blockTag}`);
+    }
+  }
+
+  type(): ChainQueryType {
+    return ChainQueryType.EthCall;
+  }
+
+  serialize(): Uint8Array {
+    const writer = new BinaryWriter()
+      .writeUint32(this.blockTag.length)
+      .writeUint8Array(Buffer.from(this.blockTag))
```

Comment:
This wont error but is this def what we want? vs creating a buffer from the hex value
```
> const x = "0xdeadbeef"
undefined
> Buffer.from(x)
<Buffer 30 78 64 65 61 64 62 65 65 66>
> const y = "deadbeef"
undefined
> Buffer.from(y, 'hex')
<Buffer de ad be ef>
```

Reply:
Yeah, this is because the block tag can be a string like "latest" for example in the future